tests/client/client_tests.py

```diff
@@ -246,11 +247,11 @@ def test_send_remote_failover_sync(should_try, sending_elasticapm_client, caplog
     should_try.return_value = True
 
     # test error
-    with caplog.at_level("ERROR", "elasticapm.transport"):
+    with capture_from_logger(caplog, logging.ERROR, "elasticapm.transport") as records:
         sending_elasticapm_client.capture_message("foo", handled=False)
     sending_elasticapm_client._transport.flush()
```

Comment:
Perhaps sticking this flush in the `with` block would solve the issues with caplog? It would let the thread catch up, right? If that works, it would probably work for both failing tests.